Add /latex-cheatsheet

rattus-rattus-rattus opened the pull request:
I have written a little custom CSS for styling some of the elements on this page. Happy to destroy them / incorporate them into the main CSS stuff.

jkynaston committed:
init

jkynaston committed:
Merge branch 'main' into 4-adapt-latex-cheatsheet

jkynaston committed:
Add /latex-cheatsheet

Includes some custom css styling

mniip commented on _layouts/cheatsheet.html:
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
```
I do not appreciate the copy-paste. Factor out the common stuff? Maybe we could instead have a `page.extra_styles` property?

mniip replied:
Maybe like this: https://github.com/discord-math/discord-math.github.io/pull/10

mniip commented on latex-cheatsheet.md:
```diff
@@ -0,0 +1,281 @@
+  - [Calculus](#calculus)
+  - [Simple geometry](#simple-geometry)
+- [Further resources](#further-resources)
+<hr>
```
Replace with
```md
## Table of Contents
{:.no_toc}

* TOC
{:toc}
```

mniip commented on latex-cheatsheet.md:
```diff
@@ -0,0 +1,281 @@
+## What is $\LaTeX$?
+$\LaTeX$ is the standard system used for rendering mathematical and scientific documents. It is also how you can render mathematical expressions and formulae server-wide using the [@TeXiT](https://discord.com/users/510789298321096704){: .mention target="_blank"} bot. If you've ever wondered how scientific papers or textbooks are laid out in the way they are, this is the guide for you.
+
+Unlike other document creation systems such as Microsoft Word, documents produced using $\LaTeX$ are not written using a so-called "what-you-see-is-what-you-get" format. Instead, documents are written in plaintext `.tex` files in the $\LaTeX$ language that are then compiled into `.pdf` documents.
```
Wouldn't mind an `<em>` on the `compiled`

mniip replied:
Actually why not use markdown's builtin `*text*` syntax?

mniip commented on latex-cheatsheet.md:
```diff
@@ -0,0 +1,281 @@
+  - `\` - The backslash symbol used for "commands" or "macros" which insert special symbols or notation into the text.
+  - `{ expr }` - Braces are used both to pass arguments to macros and to group symbols/expressions into a block.
+  - `$ expr $` or `\( expr \)` - These symbols render the enclosed expressions in <i>inline math mode</i> (or simply math mode).
+  - `$$ exp $$` or `\[ expr \]` - These symbols render the enclosed expressions in <i>display math mode</i> (or simply display mode). We will see the difference between the two math modes [below](#examples).
```
Not a big fan of this paragraph overall. I prefer the original version.

Braces are often not *required* to pass arguments because the macro is not defined with braces, the original version had a footnote about `\frac12`. Sure it is bad style, but the same happens with sub/superscripts, so it makes sense to emphasize that what they do is grouping. Whether a macro requires its argument to be grouped is a secondary concern.

The original paragraph (actually 2 paragraphs) made an introduction that is more gentle: first we suggest that formulae belong to their own "mode", and then we specify how to invoke inline or display.

Also: prefer `*text*` over `<i>`

mniip commented on latex-cheatsheet.md:
```diff
@@ -0,0 +1,281 @@
+  - `$$ exp $$` or `\[ expr \]` - These symbols render the enclosed expressions in <i>display math mode</i> (or simply display mode). We will see the difference between the two math modes [below](#examples).
+  - Note that there are two ways to enter both math modes. While both versions operate in largely the same way, there are technical reasons to prefer `\(` and `\[` over their dollar sign equivalents.
+
+The majority of other characters (namely `!`, `'`, `(`, `)`, `*`, `+`, `,`, `-`, `.`, `/`, `:`, `;`, `<`, `=`, `>`, `?`, `[`, `]`, and `|`) are rendered as themselves. Note that certain characters, such as the braces `{` and `}`, must be <i>escaped</i> using the backslash symbol (i.e. `\{` and `\}`) to be rendered correctly as they are otherwise reserved by the $\LaTeX$ compiler.
```
The original document did not go into details of "escaping" and "reserved by" (I think these are fairly technical terms that may be unfamiliar to non-programmers). Once again I like the paragraph from the original document more.

mniip commented on latex-cheatsheet.md:
````diff
@@ -0,0 +1,281 @@
+### Inline vs Display mode
+Let's look at a $\LaTeX$ snippet that renders an equation in math mode:
+```latex
+Perhaps the most famous equation in all of physics is \(e=mc^2\)
````
E not e

mniip commented on latex-cheatsheet.md:
```diff
@@ -0,0 +1,281 @@
+</div>
+
+## Using $\LaTeX$ on the server
+The <a class="mention">@TeXiT</a> bot can be used anywhere on the server to render mathematics using $\LaTeX$. To use it, simply write a normal message with any latex expressions that you want to be rendered enclosed in either `$ ... $` or `$$ ... $$`.
```
Technically it's anything that contains balanced `$ $`, `$$ $$`, `\( \)` or `\[ \]`. It's a little odd that above you recommend `\(` over `$` but here you do not.

mniip commented on latex-cheatsheet.md:
```diff
@@ -0,0 +1,281 @@
+## Using $\LaTeX$ on the server
+The <a class="mention">@TeXiT</a> bot can be used anywhere on the server to render mathematics using $\LaTeX$. To use it, simply write a normal message with any latex expressions that you want to be rendered enclosed in either `$ ... $` or `$$ ... $$`.
+
+## Command reference
```
A common mistake is using `sin` instead of `\sin`. Maybe list those somewhere?

mniip reviewed (changes requested):
You've added a lot of prose to comment on the various design details and pitfalls which makes this no longer feel like a cheatsheet. Also not a big fan of the writing style with "Nice!" and "Yikes!"

mniip committed:
Allow extra CSS to be specified per page

jkynaston committed:
Merge remote-tracking branch 'origin/extra_styles' into 4-adapt-latex-cheatsheet

jkynaston committed:
changes in response to feedback

revert some paragraphs to their original source. refactor to use extra_styles css.

jkynaston committed:
fix unfinished section
